pkg/controller/machine.go

```diff
@@ -245,16 +259,39 @@ func (c *controller) addNodeToMachine(obj interface{}) {
 		return
 	}
 
-	klog.V(4).Infof("Add machine object backing node %q", machine.Name)
-	c.enqueueMachine(machine)
+	if machine.Status.CurrentStatus.Phase != v1alpha1.MachineCrashLoopBackOff && nodeConditionsHaveChanged(machine.Status.Conditions, node.Status.Conditions) {
```

Comment:
`addNodeToMachine()` is also called when a node joins. If a new machine has been in crash loop and then becomes healthy, we will not enqueue the machine immediately when the node joins because of this check, right?

Reply:
Hm, I think it's safe to simply remove the check for `MachineCrashLoopBackOff` here, as it's unlikely to have node-object when creation itself fails.

Reply:
it is also called from `updateNodeToMachine()`. Still it should be safe to remove the check

Reply:
Yes, you need this check else the `updateNodeToMachine()` has to be modified. Also during node joining the nodeConditions do change as machineStatus conditions are initially nil.

Reply:
@ggaurav10 @prashanth26 I made the changes, can you please take another look.